python/py-pytorch/Portfile

```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem                                      1.0
+PortGroup           python                      1.0
+PortGroup           cxx11                       1.1
+PortGroup           github                      1.0
+PortGroup           mpi                         1.0
+PortGroup           compiler_blacklist_versions 1.0
+
+name                py-pytorch
+version             1.0.1
+revision            0
+github.setup        pytorch pytorch ${version} v
+
+fetch.type          git
+post-fetch {
+    system -W ${worksrcpath} "git submodule update --init"
+}
+
+platforms           darwin
+supported_archs     x86_64
+
+license             BSD
+
+maintainers         {jonesc @cjones051073} openmaintainer
+
+description         Tensors and dynamic neural networks in Python with strong GPU acceleration
+long_description    ${description}
+
+homepage            https://pytorch.org/
+
+github.livecheck.regex  {([0-9.]+)}
+
+# Support python versions
+python.versions     27 35 36 37
```

Comment:
I would only have py27 and py37. In other words, only add the latest python versions.

Reply:
why ? Pretty much all python ports have multiple py3x versions. If its a general policy to start dropping all but 37 then fine, but until then I would prefer to offer the same alternatives other ports do.

Reply:
They all do because they were added when older versions were available and then newer pythons get added as they come along. No need to go back and add the old versions of python when creating new python ports unless there is a specific need for the older versions because it requires more maintenance and testing to have all of these old python versions kept around.

Reply:
You are the maintainer of these proposed python modules, so it's your call.

Reply:
I don't disagree that dropping older python versions might make sense. However, if we do that it should in my view be done centrally, i.e. we drop *all* e.g. py35  versions across *all* ports at the same time. Not piecemeal.

Also note that at this time not all ports offer py37 versions. So only have py37 here would mean potentially for some users if they needed one python port that only had py37, and another that did not have py37, they would not be able to have both at the same time in the same python runtime which would not be good. To that end I think it is a bit  premature to drop py36 at this time.

I think I will leave as is (note I have already dropped py34, which many ports still offer) but if at some point someone proposed an across-the board dropping of older python versions, I would not be against that.

Reply:
Thank you for considering and discussing this. It's pretty complex trying to remove the old python versions. But this is probably not the right place to continue this dialogue.